### PR TITLE
Refactor `PeoplePane.test.tsx` and drop dependence on `COMMUNICATION_REACT_FLAVOR` envvar

### PR DESCRIPTION
**Pull request by prprabhu-ms**
The tests were very hard to read due to the difference branches within the test. This PR splits the tests into 3 suites for the three distinct UX's being tested - (1) flyout, (2) side pane (3) full screen with drawer.

It also replaces the use of `COMMUNICATION_REACT_FLAVOR` runtime check with the relevant conditional compilation directive.

There are some tangential changes to the UI snapshots because this PR unifies the fake remote participants used by the tests (the differences were not significant to the tests) to reduce boilerplate.

### CODE REVIEW

File: packages/react-composites/tests/browser/call/hermetic/PeoplePane.test.ts [hunk 1]
```diff
@@ -9,185 +9,209 @@ import {
...
-      await pageClick(page, dataUiId(IDS.participantButtonPeopleMenuItem));
-      await page.hover(dataUiId('participant-item') + ' >> nth=0');
-      // There should be at least one participant. Just click on the first.
-      await pageClick(page, dataUiId(IDS.participantItemMenuButton) + ' >> nth=0');
-
-      const injectedMenuItem = await waitForSelector(page, dataUiId('test-app-participant-menu-item'));
-      await injectedMenuItem.waitForElementState('stable', { timeout: perStepLocalTimeout() });
-    }
+    await pageClick(page, dataUiId('call-with-chat-composite-more-button'));
+    const drawerPeopleMenuDiv = await page.$('div[role="menu"] >> text=People');
```
Comment (dmceachernmsft) [inline packages/react-composites/tests/browser/call/hermetic/PeoplePane.test.ts:195]: Would there be any value in making a DataUIId for this button?

Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/call/hermetic/PeoplePane.test.ts:195]: I'll refrain from adding that in this PR.

I think there would be value in adding a helper function for this. Raw `$()` expressions are hard to read in a test.

File: packages/react-composites/tests/browser/call/hermetic/PeoplePane.test.ts [hunk 2]
```diff
@@ -9,185 +9,209 @@ import {
...
+const participantListInitialState = (): MockCallAdapterState => {
+  const paul = defaultMockRemoteParticipant('Paul Bridges');
+  addVideoStream(paul, true);
+  paul.isSpeaking = true;
+  const initialState = defaultMockCallAdapterState([
+    paul,
+    defaultMockRemoteParticipant('Eryka Klein'),
+    defaultMockRemoteParticipant('Fiona Harper')
+  ]);
+  addDefaultMockLocalVideoStreamState(initialState);
```
[the remark this answers is not in the document]
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/call/hermetic/PeoplePane.test.ts:215]: We do: 
image

The green boxes are the videos (with a moving bus hidden behind the green mask ;) )

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts [hunk 3]
```diff
@@ -7,16 +7,15 @@ import {
...
   isTestProfileDesktop,
-  isTestProfileStableFlavor,
   pageClick,
   stableScreenshot,
   waitForSelector
 } from '../../common/utils';
 import { buildUrlWithMockAdapter, defaultMockCallAdapterState, defaultMockRemoteParticipant, test } from './fixture';
 
 test.describe('Participant pane tests', async () => {
+  /* @conditional-compile-remove(one-to-n-calling) */
```
Comment (edwardlee-msft) [inline packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts:17]: nit: Possible for another PR, but this should include /* @conditional-compile-remove(PSTN-calls) */ as well.

**Review by edwardlee-msft — APPROVED** [review]

File: packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts (hunk 3 above)
Comment (prprabhu-ms) [reply packages/react-composites/tests/browser/call/hermetic/ParticipantPane.test.ts:17]: Oh yeah, sorry I should have rebased before tagging you. It's fixed in `main` from an earlier PR.

**Pull request merged**